eos/utils/omnes-factor-impl.hh

```diff
@@ -0,0 +1,307 @@
+/*
+ * Copyright (c) 2024 Florian Herren
+ *
+ * This file is part of the EOS project. EOS is free software;
+ * you can redistribute it and/or modify it under the terms of the GNU General
+ * Public License version 2, as published by the Free Software Foundation.
+ *
+ * EOS is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc., 59 Temple
+ * Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+
+#ifndef EOS_GUARD_EOS_UTILS_OMNES_FACTOR_IMPL_HH
+#define EOS_GUARD_EOS_UTILS_OMNES_FACTOR_IMPL_HH 1
+
+#include <eos/maths/complex.hh>
+#include <eos/maths/legendre-polynomial-vector.hh>
+#include <eos/utils/exception.hh>
+#include <eos/utils/omnes-factor.hh>
+
+#include <gsl/gsl_blas.h>
+#include <gsl/gsl_linalg.h>
+#include <gsl/gsl_matrix.h>
+
+#include <array>
+
+namespace eos
+{
+
+    // Allocate memory and initialise zeros & weights
+    template <unsigned order_, unsigned nints_>
+    OmnesFactor<order_, nints_>::OmnesFactor(std::array<double, nints_> & intervals) :
+        _intervals(intervals),
+        _sys(gsl_matrix_calloc(nints_*order_ + 1,nints_*order_)),
+        _sys2(gsl_matrix_calloc(nints_*order_ + 1,nints_*order_)),
+        _V(gsl_matrix_calloc(nints_*order_,nints_*order_)),
+        _S(gsl_vector_calloc(nints_*order_)),
+        _work(gsl_vector_calloc(nints_*order_)),
+        _x(gsl_vector_calloc(nints_*order_)),
+        _b(gsl_vector_calloc(nints_*order_+1))
+    {
+        // Perform pointer check
+        if (_sys == nullptr)
+            throw std::bad_alloc();
+        if (_sys == nullptr)
```

Comment:
1. ``_sys`` checked twice.
2. (coding style) Short if statement can appear on the same line, but still require braces, e.g., ``if (...) { throw std::bad_alloc(); }``.
3. I think it would suffice throwing ``bad_alloc`` in case that any of the allocations failed, instead of each. So maybe ``if ((_sys == nullptr) || (_V == nullptr) || ...) { throw std::bad_alloc(); }``

Reply:
I think that this one was directly taken from my kmatrix implementation, so I am the culprit!

Reply:
Good catch, the second one should be _sys2...